datastore/google/cloud/datastore/entity.py

```diff
@@ -173,6 +173,11 @@ def __ne__(self, other):
         """
         return not self == other
 
+    def __setitem__(self, key, value):
```

Comment:
It seems odd having this one change from the default serialization into the dict. We have a `def entity_to_protobuf(entity):` method. Is this the correct place to do this?

Reply:
I think, in issue they were speaking only about converting fields of entity, like `employee_entity['type'] == EmployeeType.PART_TIME`, while `entity_to_protobuf` converts the whole entity

Reply:
https://github.com/googleapis/google-cloud-python/blob/master/datastore/google/cloud/datastore/helpers.py#L216 already walks the entity to convert types. Could this be done there?

Reply:
[setitem](https://docs.python.org/3/reference/datamodel.html#object.__setitem__) will be called every time user will try to put value into field, like
`entity['field'] = some_enum`
And enum's `int` representation will be stored. So `entity_to_protobuf` don't really need any changes, 'cause values will already be ints when you'll call `entity_to_protobuf`

But if we'll move this changes to `entity_to_protobuf`, user will be storing pure enums into entity fields, and if he'll try to get field value like:
`entity['field']`
He'll got a enum, instead of `int`. And he wants to get `int` in a first place